spec/Brille24/SyliusCustomerOptionsPlugin/Updater/CustomerOptionPriceUpdaterSpec.php

```diff
@@ -0,0 +1,569 @@
+            $customerOption,
+            $customerOptionValue,
+            $channel,
+            $valuePrice,
+            $product,
+            $customerOptionCode,
+            $customerOptionValueCode,
+            $channelCode,
+            $productCode,
+            $validFrom,
+            $validTo,
+            $type,
+            $amount,
+            $percent
+        );
+        $customerOptionValuePriceFactory->createNew()->shouldBeCalled()->willReturn($valuePrice);
+        $valuePrice->setCustomerOptionValue($customerOptionValue)->shouldBeCalled();
+        $valuePrice->setChannel($channel)->shouldBeCalled();
+        $valuePrice->setProduct($product)->shouldBeCalled();
+
+        $customerOptionValuePriceRepository->findBy([
+            'customerOptionValue' => $customerOptionValue,
+            'channel'             => $channel,
+            'product'             => $product,
+        ])->shouldBeCalled()->willReturn([]);
+
+        $valuePrice->setProduct($product)->shouldBeCalled();
+        $valuePrice->setChannel($channel)->shouldBeCalled();
+        $valuePrice->setCustomerOptionValue($customerOptionValue)->shouldBeCalled();
+        $valuePrice->setPercent($percent)->shouldBeCalled();
+        $valuePrice->setAmount($amount)->shouldBeCalled();
+        $valuePrice->setType($type)->shouldBeCalled();
+
+        $product->getCustomerOptionValuePrices()->willReturn(new ArrayCollection());
+        $validator->validate(
+            Argument::type(Collection::class),
+            Argument::type(CustomerOptionValuePriceDateOverlapConstraint::class)
+        )->shouldBeCalled()->willReturn([$violation]);
+
+        $this->shouldThrow(\InvalidArgumentException::class)->during('updateForProduct', [
+            $customerOptionCode,
+            $customerOptionValueCode,
+            $channelCode,
+            $product,
+            $validFrom,
+            $validTo,
+            $type,
+            $amount,
+            $percent,
+        ]);
+    }
+
+    /**
+     * @param CustomerOptionRepositoryInterface $customerOptionRepository
+     * @param RepositoryInterface $customerOptionValuePriceRepository
+     * @param CustomerOptionValueRepositoryInterface $customerOptionValueRepository
+     * @param ChannelRepositoryInterface $channelRepository
+     * @param ProductRepositoryInterface $productRepository
+     * @param CustomerOptionInterface $customerOption
+     * @param CustomerOptionValueInterface $customerOptionValue
+     * @param ChannelInterface $channel
+     * @param CustomerOptionValuePriceInterface $valuePrice
+     * @param ProductInterface|null $product
+     * @param string $customerOptionCode
+     * @param string $customerOptionValueCode
+     * @param string $channelCode
+     * @param string $productCode
+     * @param string $validFrom
+     * @param string $validTo
+     * @param string $type
+     * @param int $amount
+     * @param float $percent
+     */
+    private function setupMocks(
```

Comment:
Maybe you can call / integrate this function to be run in the let function to set up all the mocks before the test has started.